# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/memtest86/hotfix-x32-64.patch
# Copyright (C) 2022 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- memtest86plus-6.00/build64/Makefile.vanilla	2022-11-25 21:45:51.747494223 +0100
+++ memtest86plus-6.00/build64/Makefile	2022-11-25 21:57:17.125459700 +0100
@@ -1,4 +1,4 @@
-AS = as -64
+AS = as
 CC = gcc
 
 GIT = git
@@ -9,8 +9,10 @@
   GIT_AVAILABLE = true
 endif
 
+ASFLAGS = -64
+
 CFLAGS = -std=c11 -Wall -Wextra -Wshadow -m64 -march=x86-64 -mno-mmx -mno-sse -mno-sse2 \
-         -fpic -fno-builtin -ffreestanding -fomit-frame-pointer -fno-stack-protector
+         -fpic -fno-builtin -ffreestanding -fomit-frame-pointer -fno-stack-protector -m64
 
 INC_DIRS = -I../boot -I../system -I../lib -I../tests -I../app -Iapp
 
@@ -75,7 +77,7 @@
 -include $(subst .o,.d,$(APP_OBJS))
 
 boot/%.o: boot/%.s
-	$(AS) $< -o $@
+	$(AS) $(ASFLAGS) $< -o $@
 
 boot/startup.s: ../boot/startup64.S ../boot/boot.h
 	@mkdir -p boot
